tests/conftest.py

```diff
@@ -91,6 +91,23 @@ def rain_data_folder(tmp_path):
     return example_rain_path
 
 
+@pytest.fixture()
+def rain_data_folder_non_existing(tmp_path):
```

Comment:
The implementation is correct. I actually would propose to either:
- integrate the code in the tests itself (and use the tmp_path fixture as input in the unit test) if you only need the non-existing or empty folder once
- use a more generic name for these fixtures instead of rainfall so it can be reused: `data_folder_non_existing` and `data_folder_empty`

My proposal is the second option.

Reply:
Adjusted according to option 2. I follow that it is more optimal for extension